src/controllers/shared.controllers/director.confirm.correspondence.address.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { NextFunction, Request, Response } from "express";
+import {
+  DIRECTOR_CORRESPONDENCE_ADDRESS_MANUAL_PATH,
+  UPDATE_DIRECTOR_CORRESPONDENCE_ADDRESS_MANUAL_PATH,
+} from "../../types/page.urls";
+import { urlUtils } from "../../utils/url";
+import { Session } from "@companieshouse/node-session-handler";
+import { getOfficerFiling, patchOfficerFiling } from "../../services/officer.filing.service";
+import { formatTitleCase } from "../../utils/format";
+import { OfficerFiling } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
+import { getDirectorNameBasedOnJourney, setBackLink } from "../../utils/web";
+
+
+export const getConfirmCorrespondence = async (req: Request, res: Response, next: NextFunction, templateName: string, backUrlPath: string, isUpdate?: boolean) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+    const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+    let enterAddressManuallyUrl: string;
+
+    if(isUpdate){
+      enterAddressManuallyUrl = urlUtils.getUrlToPath(UPDATE_DIRECTOR_CORRESPONDENCE_ADDRESS_MANUAL_PATH, req)
+    } else {
+      enterAddressManuallyUrl = urlUtils.getUrlToPath(DIRECTOR_CORRESPONDENCE_ADDRESS_MANUAL_PATH, req)
+  }
+
+    return res.render(templateName, {
+      templateName: templateName,
+      backLinkUrl: setBackLink(req, officerFiling.checkYourAnswersLink,urlUtils.getUrlToPath(backUrlPath, req)),
```

Comment:
do we need to go back to the CYA page? I believe with the flow change user will always go back to the lookup page.

Reply:
Yeah that makes sense, will update this now. Thanks!